### PR TITLE
Change database schema for capability status

### PR DESCRIPTION
**Pull request by andreasfrisch**
closes dfds/cloudplatform/issues/1851

# Warning
Modifies existing database table and transforms the data
There is no going back from a failure

# Additional Review Notes
I have tested this locally, running the migration script against the database produced by the develop-branch using pgAdmin.
This PR is constrained to introducing the capability status and making sure everything works as expected.

Bonus: I changed the seeding scripts so that the 4 main contributors of this repository are automatically members of the dummy-capabilities present during local development

### CODE REVIEW

File: db/migrations/20230801113000_add-capability-status.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+SET
+    "Status" = 'Deleted',
+    "ModifiedAt" = "Deleted"
+WHERE "Deleted" IS NOT NULL;
+
+UPDATE "Capability"
+SET "ModifiedAt" = "CreatedAt"
+WHERE "Deleted" IS NULL;
+
+-- remove deleted coloumn
```
Comment (kralle333) [inline db/migrations/20230801113000_add-capability-status.sql:20]: spelling mistake ;)

File: db/migrations/20230801113000_add-capability-status.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ALTER TABLE "Capability"
+DROP column "Deleted";
```
Comment (kralle333) [inline db/migrations/20230801113000_add-capability-status.sql:22]: There is mixed casing in the migration, can we make it consistent? I have chosen all uppercase in confluent gateway, so maybe we want this here too?

File: db/seed/Member.csv [hunk 3]
```diff
@@ -1,3 +1,5 @@
 Email;DisplayName;Id
-jandr@dfds.com;Jens Andresen;jandr@dfds.com
-thfis@dfds.com;Thomas Fischer;thfis@dfds.com
+andfris@dfds.com;Andreas Frisch;andfris@dfds.com
+krijens@dfds.com;Kristian Pilegaard Jensen;krijens@dfds.com
+pausegh@dfds.com;Paul Seghers;pausegh@dfds.com
+hritote@dfds.com;Hristiyana Toteva;hritote@dfds.com
```
Comment (kralle333) [inline db/seed/Member.csv:5]: I think actually Emil was not happy to have real emails in our test data in another repo. Can you make up some emails?

File: db/seed/Membership.csv [hunk 4]
```diff
@@ -1,3 +1,10 @@
...
-marketing-department-xxx;thfis@dfds.com;2023-04-04T16:00:00;C20C9307-55E0-4039-B034-9E061499D7B5
+cloudengineering-xxx;andfris@dfds.com;2023-03-17T08:00:00;35C66428-811E-41E2-9B94-8E09995A0367
+pending-deletion-xxx;andfris@dfds.com;2023-03-17T08:00:00;35C66428-811E-41E2-9B94-8E09995A0368
+cloudengineering-xxx;krijens@dfds.com;2023-03-17T08:00:00;cf1db68c-3073-11ee-be56-0242ac120002
+pending-deletion-xxx;krijens@dfds.com;2023-03-17T08:00:00;cf1db68c-3073-11ee-be56-0242ac120003
+cloudengineering-xxx;pausegh@dfds.com;2023-03-17T08:00:00;d40676c0-3073-11ee-be56-0242ac120002
+pending-deletion-xxx;pausegh@dfds.com;2023-03-17T08:00:00;d40676c0-3073-11ee-be56-0242ac120003
+cloudengineering-xxx;hritote@dfds.com;2023-03-17T08:00:00;daf5f5fa-3073-11ee-be56-0242ac120002
+pending-deletion-xxx;hritote@dfds.com;2023-03-17T08:00:00;daf5f5fa-3073-11ee-be56-0242ac120003
+marketing-department-xxx;andfris@dfds.com;2023-04-04T16:00:00;C20C9307-55E0-4039-B034-9E061499D7B5
```
Comment (kralle333) [inline db/seed/Membership.csv:10]: same comment as for Member.csv

File: src/SelfService/Domain/Models/Capability.cs [hunk 5]
```diff
@@ -8,17 +8,17 @@ public class Capability : AggregateRoot<CapabilityId>
...
     )
         : base(id)
     {
         Name = name;
         Description = description;
-        Deleted = deleted;
         CreatedAt = createdAt;
+        Status = CapabilityStatusOptions.Active; // all new capabilities are active to begin with
         CreatedBy = createdBy;
+        ModifiedAt = createdAt; // this will always be the same as CreatedAt for a new Capability
```
Comment (kralle333) [inline src/SelfService/Domain/Models/Capability.cs:21]: In continuation of our discussion of how and when to add comments. In this scenario I think you should omit them, for me it is obvious from the code, but up to you :)

File: db/migrations/20230801113000_add-capability-status.sql [hunk 6]
```diff
@@ -0,0 +1,22 @@
+-- 2023-08-01 11:30:xx : Add status and modified_at to capabilities
+
+-- add columns
+ALTER TABLE "Capability"
+ADD column "Status" varchar(255) NOT NULL DEFAULT 'Active',
+ADD column "ModifiedAt" timestamp NOT NULL DEFAULT clock_timestamp();
```
Comment (kralle333) [inline db/migrations/20230801113000_add-capability-status.sql:6]: Just curious: are these timestamps in UTC time?

**Review by kralle333 — APPROVED** [review]
Great! Nice to get rid of that weird deleted column

File: db/migrations/20230801113000_add-capability-status.sql (hunk 1 above)
Comment (andreasfrisch) [reply db/migrations/20230801113000_add-capability-status.sql:20]: not at all ;)

File: db/seed/Member.csv (hunk 3 above)
Comment (andreasfrisch) [reply db/seed/Member.csv:5]: But these are linked to our dfds-ad accounts. If this is to work with AD, they must be real

File: db/migrations/20230801113000_add-capability-status.sql (hunk 6 above)
Comment (andreasfrisch) [reply db/migrations/20230801113000_add-capability-status.sql]: They are not, but that is not truly relevant here.
We must have a default value, but it is never used due to the rest of the code.
Added comment around this.

**Pull request merged**